discovery-infra/tests/test_wrong_boot_order.py

```diff
@@ -20,12 +25,165 @@ def test_wrong_boot_order_one_node(self, nodes, cluster):
 
         # Wait until wrong boot order
         new_cluster.wait_for_one_host_to_be_in_wrong_boot_order()
+        new_cluster.wait_for_cluster_to_be_in_installing_pending_user_action_status()
 
-        # Reboot required nodes into ISO
+        # Reboot required nodes into HD
         node.shutdown()
         node.set_boot_order(cd_first=False)
         node.start()
 
         # wait until all nodes are in Installed status, will fail in case one host in error
+        new_cluster.wait_for_cluster_to_be_in_installing_status()
+        new_cluster.wait_for_hosts_to_install()
+        new_cluster.wait_for_install()
+
+    @pytest.mark.regression
+    def test_installation_succeeded_after_all_nodes_have_incorrect_boot_order(self,
+                                                                              nodes,
+                                                                              cluster):
+        # Define new cluster
+        new_cluster = cluster()
+
+        # Change boot order of all the nodes
+        for n in nodes:
+            n.set_boot_order(cd_first=True)
+
+        # Start cluster install
+        new_cluster.prepare_for_install(nodes)
+        new_cluster.start_install()
+        new_cluster.wait_for_installing_in_progress()
+
+        # Wait until wrong boot order - all hosts except bootstrap
+        new_cluster.wait_for_hosts_to_be_in_wrong_boot_order(len(nodes)-1)
+        new_cluster.wait_for_cluster_to_be_in_installing_pending_user_action_status()
```

Comment:
What is the behavior when cluster moves to installing pending user action? 
in such state installation continues? 
cuase here you wait for all hosts to have wrong boot order, so when does the cluster status changes? after first wrong boot node?

Reply:
resolved since we discussed it in private.